darts/models/forecasting/regression_ensemble_model.py

```diff
@@ -100,34 +115,155 @@ def __init__(
         )
 
         self.regression_model: RegressionModel = regression_model
+
+        raise_if(
+            regression_train_n_points == -1
+            and not (self.all_trained and (not retrain_forecasting_models)),
+            "`regression_train_n_points` can be set to `-1` only if `retrain_forecasting_model=False` and "
+            "all the `forecasting_models` are already fitted.",
+            logger,
+        )
+
         self.train_n_points = regression_train_n_points
 
+        raise_if(
+            train_using_historical_forecasts and not self.is_global_ensemble,
+            "`train_using_historical_forecasts=True` is available only when all the models contained in "
+            "`forecasting_models` are global.",
+            logger,
+        )
+
+        self.train_using_historical_forecasts = train_using_historical_forecasts
+
     def _split_multi_ts_sequence(
         self, n: int, ts_sequence: Sequence[TimeSeries]
     ) -> Tuple[Sequence[TimeSeries], Sequence[TimeSeries]]:
         left = [ts[:-n] for ts in ts_sequence]
         right = [ts[-n:] for ts in ts_sequence]
         return left, right
 
-    def fit(
+    def _make_multiple_historical_forecasts(
         self,
-        series: Union[TimeSeries, Sequence[TimeSeries]],
+        train_n_points: int,
+        series: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
         past_covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
         future_covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+        num_samples: int = 1,
+        predict_likelihood_parameters: bool = False,
+    ) -> Union[TimeSeries, Sequence[TimeSeries]]:
+        """
+        For GlobalForecastingModel, when predicting n > output_chunk_length, `historical_forecasts()` generally
+        producebetter forecasts than `predict()`.
+
+        train_n_points are generated, starting from the end of the series
+        """
+        is_single_series = isinstance(series, TimeSeries) or series is None
+        predictions = [
+            model.historical_forecasts(
```

Comment:
I decided to go with `forecast_horizon = stride = output_chunk_length` and `last_points_only=False`. When `output_chunk_length` is not a multiple of `regression_train_n_points`, the historical forecast is missing some values at the end (using a stride of 1 could solve the problem but then, there is the problem of the overlapping forecasts) so I added exception in the constructor to prevent this scenario. Let me know what you think.